PWG/EMCAL/EMCALbase/AliAnalysisTaskEmcalEmbeddingHelperData.cxx

```diff
@@ -0,0 +1,175 @@
+#include <vector>
+
+#include "AliAnalysisTaskEmcalEmbeddingHelper.h"
+#include "AliAnalysisTaskEmcalEmbeddingHelperData.h"
+#include "AliAODEvent.h"
+#include "AliESDEvent.h"
+#include "AliVAODHeader.h"
+#include "AliEmcalList.h"
+#include "AliAnalysisManager.h"
+
+/// \cond CLASSIMP
+ClassImp(AliAnalysisTaskEmcalEmbeddingHelperData);
+/// \endcond
+
+AliAnalysisTaskEmcalEmbeddingHelperData::AliAnalysisTaskEmcalEmbeddingHelperData():
+  AliAnalysisTaskEmcalEmbeddingHelper(),
+  fPtSelection()
+{
+}
+
+
+AliAnalysisTaskEmcalEmbeddingHelperData::AliAnalysisTaskEmcalEmbeddingHelperData(const char *name):
+  AliAnalysisTaskEmcalEmbeddingHelper(name),
+  fPtSelection()
+{
+}
+
+AliAnalysisTaskEmcalEmbeddingHelperData::~AliAnalysisTaskEmcalEmbeddingHelperData()
+{
+}
+
+void AliAnalysisTaskEmcalEmbeddingHelperData::UserCreateOutputObjects() 
+{
+
+  // run function from embedding helper base class
+  AliAnalysisTaskEmcalEmbeddingHelper::UserCreateOutputObjects();
+
+  TString histName = "fHistPtSelection";
+  TString histTitle = "pT selection;Result;p_{T}^{lead}";
+  auto histPtSelection= fHistManager.CreateTH2(histName, histTitle, 2, 0, 2, 200, 0, 100);
+  histPtSelection->GetXaxis()->SetBinLabel(1,"Accepted");
+  histPtSelection->GetXaxis()->SetBinLabel(2,"Rejected");
+
+  fOutput->Add(histPtSelection);
+
+  PostData(1, fOutput);
+
+
+}
+
+/**
+ * Override inherited function and add check on track with max pT.
+ * Rejects event if track within set pT range is not found
+ */
+
+Bool_t AliAnalysisTaskEmcalEmbeddingHelperData::CheckIsEmbeddedEventSelected()
+{
+
+  if(!AliAnalysisTaskEmcalEmbeddingHelper::CheckIsEmbeddedEventSelected()) return kFALSE;
+
+  Int_t nTracks = fExternalEvent->GetNumberOfTracks();
+  Int_t nTracksInRange = 0;
+  Double_t ptMax = 0;
+  for(Int_t i=0;i<nTracks;i++) {
+    Double_t pt = fExternalEvent->GetTrack(i)->Pt();
+    if(pt>ptMax) ptMax = pt;
+    //now check set windows
+    Int_t npt = fPtSelection.size();
+    for(Int_t j=0;j<npt;j++) {
+      if(pt>fPtSelection[j].first && pt<fPtSelection[j].second) { 
+        // found track within pt
+        nTracksInRange++;
+      }
+    }
+  }
+
+  TString histName = "fHistPtSelection";
+  if(nTracksInRange==0) {
+    fHistManager.FillTH2(histName, 1.5, ptMax );
+    return kFALSE;
+  }
+  else {
+    fHistManager.FillTH2(histName, 0.5, ptMax );
+  }
+
+
+  return kTRUE;
+}
+
+
+void AliAnalysisTaskEmcalEmbeddingHelperData::RetrieveTaskPropertiesFromYAMLConfig() {
+
+  AliAnalysisTaskEmcalEmbeddingHelper::RetrieveTaskPropertiesFromYAMLConfig();
+
+  // get pt selection from YAML file
+  // defined in 2 arrays, pt
+
+  std::vector <double> ptSelectionMin;
```

Comment:
That all looks good. To simplify, you might try loading directly into the object (I didn't test this here, but I'm done similar in other places):

```yaml
ptSelection: [[1, 2], [3,4]]
```

and then load it with

```c++
bool res = fYAMLConfig.GetProperty("ptSelection", fPtSelection);
```

But feel free to take it or leave it